src/osip-simulation-controller/src/main/java/edu/kit/pse/osip/simulation/controller/AbstractTankServer.java

```diff
@@ -1,72 +1,103 @@
 package edu.kit.pse.osip.simulation.controller;
 
+import edu.kit.pse.osip.core.io.networking.UAIdentifiers;
+import edu.kit.pse.osip.core.opcua.server.UAServerWrapper;
+import edu.kit.pse.osip.core.utils.language.Translator;
+import org.eclipse.milo.opcua.stack.core.Identifiers;
+import org.eclipse.milo.opcua.stack.core.UaException;
+import org.eclipse.milo.opcua.stack.core.types.builtin.DataValue;
+import org.eclipse.milo.opcua.stack.core.types.builtin.Variant;
+
 /**
- * Allows setting the values inside the OPC UA server without having to think about NodeIds, namespaces or data types. Provides concrete methods for the variables used in our simulation.
+ * Allows setting the values inside the OPC UA server without having to think about NodeIds,
+ * namespaces or data types. Provides concrete methods for the variables used in our simulation.
+ *
+ * @author Hans-Peter Lehmann
+ * @version 1.0
  */
-public abstract class AbstractTankServer extends edu.kit.pse.osip.core.opcua.server.UAServerWrapper {
+public abstract class AbstractTankServer extends UAServerWrapper {
     /**
      * Represents a tank inside an OPC UA server
      * @param namespaceName The name of the namespace
      * @param port The port on which the server should be available
+     * @throws UaException If the server can not be created
      */
-    public AbstractTankServer (String namespaceName, int port) {
+    public AbstractTankServer(String namespaceName, int port) throws UaException {
         super(namespaceName, port);
-        throw new RuntimeException("Not implemented!");
+
+        Translator t = Translator.getInstance();
+        addFolder(UAIdentifiers.FOLDER_TANK, t.getString("opcua.tank"));
+        addFolder(UAIdentifiers.FOLDER_LIQUID, t.getString("opcua.liquid"));
+        addFolder(UAIdentifiers.FOLDER_LIQUID_ALARMS, t.getString("opcua.alarms"));
+        addFolder(UAIdentifiers.FOLDER_PIPE, t.getString("opcua.pipe"));
+        addFolder(UAIdentifiers.FOLDER_PIPE_OUT, t.getString("opcua.pipe.out"));
+
+        addVariable(UAIdentifiers.COLOR, t.getString("opcua.liquid.color"), Identifiers.Int32);
+        addVariable(UAIdentifiers.FILL_LEVEL, t.getString("opcua.tank.fillLevel"), Identifiers.Float);
+        addVariable(UAIdentifiers.TEMPERATURE, t.getString("opcua.liquid.temperature"), Identifiers.Float);
+
+        addVariable(UAIdentifiers.ALARM_OVERFLOW, t.getString("opcua.tank.overflowAlarm"), Identifiers.Boolean);
+        addVariable(UAIdentifiers.ALARM_UNDERFLOW, t.getString("opcua.tank.underflowAlarm"), Identifiers.Boolean);
+        addVariable(UAIdentifiers.ALARM_OVERHEAT, t.getString("opcua.tank.overheatAlarm"), Identifiers.Boolean);
+        addVariable(UAIdentifiers.ALARM_UNDERCOOL, t.getString("opcua.tank.undercoolAlarm"), Identifiers.Boolean);
+
+        addVariable(UAIdentifiers.ALARM_UNDERCOOL, t.getString("opcua.pipe.flowRate"), Identifiers.Boolean);
+
     }
     /**
      * Set the color of the liquid in this tank
```

Comment:
In Analogie zu den Alarmen sollte hier und nachfolgend "Sets" stehen oder in Analogie zu dem Kommentar bei den Alarmen "Set".